### ISSUE TITLE
How to use other effects in `Static` effects?

### ISSUE THREAD

**Issue by oberblastmeister** [haskell-effectful/effectful#57]
The documentation explains that you can run arbitrary `IO` actions using `unsafeEff_` as long as the effects are reintroduced in function that runs the effect. How can you run `effectful` effects such as `Error`, `Writer`, etc. in the same way? Is it safe to use `unsafeCoerce` to coerce the type level effects as long as they are introduced in the run function?

**Comment by arybczak** [comment]
> Is it safe to use unsafeCoerce to coerce the type level effects as long as they are introduced in the run function?

No.

Static effects are primitive, all they are supposed to be able to do is manipulation of the StaticRep and IO. If you want to reuse existing effects, you should use dynamic dispatch and `reinterpret`.

**Comment by oberblastmeister** [comment]
Could we mitigate the usage of `unsafeEff` by having a type family for static effects that says if it requires `IO` or not. Then in functions such as `evalStaticRep` we could force the introduction of `IOE`. For example something like this

```haskell
data FileSystem :: Effect

type NeedsIO :: Effect -> Bool
type family NeedsIO e

type instance NeedsIO FileSystem = 'True

deferIO :: (DispatchOf e ~ 'Static, NeedsIO e ~ 'True) => IO a -> Eff es a
deferIO = unsafeEff_

evalStaticRepDeferred :: (DispatchOf e ~ 'Static, NeedsIO e ~ 'True, IOE :> es) => StaticRep e -> Eff (e ': es) a -> Eff es a
evalStaticRepDeferred = evalStaticRep
```

**Comment by arybczak** [comment]
Hmm. Yeah, that seems like a good idea :+1:

**Comment by arybczak** [comment]
Fixed in b8c3711cbe6246bb8d583b2796d3d937d2bd9eaa.
